ext/CatalystBifurcationKitExtension/bifurcation_kit_extension.jl

```diff
@@ -0,0 +1,19 @@
+### Dispatch for BifurcationKit BifurcationProblems ###
+
+# Creates a BifurcationProblem, using a ReactionSystem as an input.
+function BK.BifurcationProblem(rs::ReactionSystem, u0_bif, ps, bif_par, args...; plot_var=nothing, record_from_solution=BK.record_sol_default, jac=true, u0=[], kwargs...)
+    
+    # Converts symbols to symbolics.
+    (bif_par isa Symbol) && (bif_par = rs.var_to_name[bif_par])
+    (plot_var isa Symbol) && (plot_var = rs.var_to_name[plot_var])
+    (u0_bif isa Vector{Pair{Symbol, Float64}}) && (u0_bif = symmap_to_varmap(rs, u0_bif))
+    (ps isa Vector{Pair{Symbol, Float64}}) && (ps = symmap_to_varmap(rs, ps))
+    (u0 isa Vector{Pair{Symbol, Float64}}) && (u0 = symmap_to_varmap(rs, u0))
+
+    # Creates NonlinearSystem.
+    conservationlaw_errorcheck(rs, vcat(ps, u0))
+    nsys = convert(NonlinearSystem, rs; remove_conserved=true, defaults=Dict(u0))
```

Comment:
What happens to existing defaults a user might have specified in building `rs`?

Reply:
If they are given in `u0`, they are overridden. If not, they should be accounted for.

Reply:
I have updated the `convert` functions slightly to help with this:
```
function Base.convert(::Type{<:NonlinearSystem}, rs::ReactionSystem; name = nameof(rs),
                      combinatoric_ratelaws = get_combinatoric_ratelaws(rs),
                      include_zero_odes = true, remove_conserved = false, checks = false,
                      default_u0 = Dict(), default_p = Dict(), defaults = _merge(Dict(default_u0), Dict(default_p)),
                      kwargs...)
    spatial_convert_err(rs::ReactionSystem, NonlinearSystem)
    fullrs = Catalyst.flatten(rs)
    remove_conserved && conservationlaws(fullrs)
    ists, ispcs = get_indep_sts(fullrs, remove_conserved)
    eqs = assemble_drift(fullrs, ispcs; combinatoric_ratelaws, remove_conserved,
                         as_odes = false, include_zero_odes)
    error_if_constraint_odes(NonlinearSystem, fullrs)
    eqs, sts, ps, obs, defs = addconstraints!(eqs, fullrs, ists, ispcs; remove_conserved)
    NonlinearSystem(eqs, sts, ps;
                    name,
                    observed = obs,
                    defaults = _merge(defaults,defs),
                    checks,
                    kwargs...)
end
```

Reply:
This shoudl also be (partially) tested in the test case (in principle I could do various combinations over overriding and not)
```julia
let 
    # Create model
    extended_brusselator = @reaction_network begin
        @species W(t) = 2.0
        @parameters k2 = 0.5
        A, ∅ → X
        1, 2X + Y → 3X
        B, X → Y
        1, X → ∅
        (k1*Y, k2), V <--> W
    end
    @unpack A, B, k1
    u0_guess = [:X => 1.0, :Y => 1.0, :V => 0.0, :W => 0.0]
    p_start = [A => 1.0, B => 4.0, k1 => 0.1]
    
    # Computes bifurcation diagram.
    BifurcationProblem(extended_brusselator, u0_guess, p_start, :B; plot_var=:V, u0 = [:V => 1.0])
    p_span = (0.1, 6.0)
    opt_newton = NewtonPar(tol = 1e-9, max_iterations = 100)
    opts_br = ContinuationPar(dsmin = 0.0001, dsmax = 0.001, ds = 0.0001,
        max_steps = 200000, nev = 2, newton_options = opt_newton,
        p_min = p_span[1], p_max = p_span[2],
        detect_bifurcation = 3, n_inversion = 4, tol_bisection_eigenvalue = 1e-8, dsmin_bisection = 1e-9)
    bif_dia = bifurcationdiagram(bprob, PALC(), 2, (args...) -> opts_br; bothside=true)
    
    # Checks computed V values are correct (Formula: V = k2*(V0+W0)/(k1*Y+k2), where Y=2*B.)
    B_vals = getfield.(bif_dia.γ.branch, :param)
    V_vals = getfield.(bif_dia.γ.branch, :x)
    @test all(V_vals .≈ 0.5*(1.0+2.0) ./ (0.1 .* 2*B_vals .+ 0.5))
    
    # Checks that the bifurcation point is correct.
    @test length(bif_dia.γ.specialpoint) == 3 # Includes start and end point.
    hopf_bif_point = filter(sp -> sp.type == :hopf, bif_dia.γ.specialpoint)[1]
    @test isapprox(hopf_bif_point.param, 1.5, atol=1e-5)

    # Tests that an error is thrown if information of conserved species is not fully provided.
    @test_throws Exception BifurcationProblem(extended_brusselator, u0_guess, p_start, :B; plot_var=:V, u0 = [:X => 1.0])
end
```

Reply:
nev = 2: are you sure you dont have more variables?

Reply:
I just copied some ContPar code that worked, I will try to remove the necessary fields (and update nev to the number of variables!)